ocaml/idl/datamodel_tracing.ml

```diff
@@ -0,0 +1,178 @@
+      ; (Set String, "processors", "The processors of the TracerProvider")
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let destroy =
+  call ~name:"destroy" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+    ~params:[(Ref _tracing, "self", "The TracerProvider to be destoryed")]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_hosts =
+  call ~name:"set_hosts" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the host field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set (Ref _host)
+        , "hosts"
+        , "Hosts the TracerProvider is to be created on"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_enabled =
+  call ~name:"set_enabled" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the enabled field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Bool
+        , "enabled"
+        , "The enabled the TracerProvider is to be set to. true meaning the \
+           TracerProvider is enabled and false that it is not"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_tags =
+  call ~name:"set_tags" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the tags field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Map (String, String)
+        , "tags"
+        , "The tags that are emitted to spans derived from this provider"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_endpoints =
+  call ~name:"set_endpoints" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the endpoints field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set String
+        , "endpoints"
+        , "The endpoints that the TracerProvider will export spans to. A URL \
+           or the string 'bugtool'. This can refer to an enpoint to the local \
+           file system"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_components =
+  call ~name:"set_components" ~in_oss_since:None ~lifecycle:[]
+    ~doc:
+      "Set the components on which the TracerProvider is to registered on. \
+       i.e. xapi, xenopds, networkd, etc"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; ( Set String
+        , "components"
+        , "The components the TracerProvider is to be set to"
+        )
+      ]
+    ~allowed_roles:_R_POOL_ADMIN ()
+
+let set_filters =
+  call ~name:"set_filters" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the filters field of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (Ref _tracing, "self", "The TracerProvider")
+      ; (Set String, "filters", "The filter the TracerProvider is to be set to")
```

Comment:
A list of filters (comma-separated). A filter (like "..") is a name that refers to an existing filter in the Tracing Library.